vitals/Main.java

```diff
@@ -1,23 +1,42 @@
 package vitals;
-
 public class Main {
+	static float  minTemperature=0.0f;
+	static float maxTemperature=45.0f;
+	static float minSoc=20.0f;
+	static float maxSoc=80.0f;
+	static float maxChargeRate=0.8f;
+	
     static boolean batteryIsOk(float temperature, float soc, float chargeRate) {
-        if(temperature < 0 || temperature > 45) {
-            System.out.println("Temperature is out of range!");
-            return false;
-        } else if(soc < 20 || soc > 80) {
-            System.out.println("State of Charge is out of range!");
-            return false;
-        } else if(chargeRate > 0.8) {
-            System.out.println("Charge Rate is out of range!");
-            return false;
-        }
-        return true;
+        return checkTemperature(temperature)&&checkSoc(soc)&&checkChargeRate(chargeRate);
+    }
+    
+    static boolean checkTemperature(float temperature)
+    {
+    	 if(temperature < minTemperature || temperature > maxTemperature) {
+             System.out.println("Temperature is out of range!");
+             return false;
+         }
+    	 return true;
     }
+static boolean checkSoc(float soc)
```

Comment:
These three functions look the same, with similar range-checks. Their tests also look very similar.
This means any change in the comparison would need to be done in three places. Addition of new parameters will mean copy-pasting these functions and their tests as well.
Make a common function to avoid the above disadvantages.